lib/oli/utils/slug.ex

```diff
@@ -115,12 +115,38 @@ defmodule Oli.Utils.Slug do
   end
 
   @doc """
-  Generates a unique slug for the table using the title provided
+  Generates a unique slug or slugs for the table using the title or titles provided
   """
+  def generate(table, titles) when is_list(titles) do
+    # Ensure no repeated titles
+    {titles, _} =
+      Enum.reduce(titles, {[], 1}, fn title, {titles, index} ->
+        slugified_title = slugify(title)
+
+        case Enum.member?(titles, slugified_title) do
```

Comment:
There is some `n^2` behavior that we should remove here.  We need to rework this `Enum. reduce` so that inside of the loop we do not make a call to `Enum.member`.  Instead, have the initial value of the reduce be `{[], MapSet.new(), 1}`, and add each unique title to the mapset as we go.

Reply:
I thought about using a MapSet but when I transform that MapSet into a list, the order is not the same as the input list. I'm going to think of a way to make it o(n) and guarantee that the order of the output list is the same one as the input one